compiler/analysis/info.go

```diff
@@ -136,6 +166,8 @@ type FuncInfo struct {
 	GotoLabel map[*types.Label]bool
 	// List of continue statements in the function.
 	continueStmts []continueStmt
+	// List of return statements in the function.
+	returnStmts []astPath
```

Comment:
If this is only used in determining if the return may block due to a defer, should this become `returnStmtsAfterFirstDefer`?

I.e. in the common case of something like:

```golang
resp, err := http.Get(...)
if err != nil {
    return err
}
defer resp.Body.Close()

/// some time later
return ...
```

The first `return` would never be blocked by the defer.

Reply:
I thought about that, but existence of goto defeats that logic. 

```go
resp, err := http.Get(...)
goto gotcha2
gotcha1:
if err != nil {
    return err
}
gotcha2:
defer resp.Body.Close()
goto gotcha1
```

Even though I suspect examples like this that also make sense would be rare, they might pop up in generated code. This could be solved with the use of control flow analysis, but that's something beyond my grasp just yet ¯\\\_(ツ)\_/¯

On a tangential note, cases like this make me wonder if there's merit to using something like https://pkg.go.dev/golang.org/x/tools/go/ssa instead of pure AST to generate code. It would certainly lead to less reader-friendly generated JavaScript, but might simplify the compiler? Or not 🤷‍♂️